### PR TITLE
Move call of onValidationChanged to callback of setState

### PR DESCRIPTION
**Pull request by ahennr**
Title says it all.

### CODE REVIEW

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx [hunk 1]
```diff
@@ -273,15 +273,15 @@ class ComparisonFilterUi extends React.Component<ComparisonFilterProps, Comparis
...
-    if (_isFunction(onValidationChanged)) {
-      onValidationChanged(validationStateNew);
-    }
-
     onFilterChange(filter);
     this.setState({
       filter,
       validateStatus: validationStateNew
-    });
+    },            () => {
```
Comment (chrismayer) [inline src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: Is this the correct indentation?

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx [hunk 2]
```diff
@@ -304,11 +304,12 @@ class ComparisonFilterUi extends React.Component<ComparisonFilterProps, Comparis
     this.setState({
       validateStatus: validationStateNew,
       operator: newOperator
+    },            () => {
```
Comment (chrismayer) [inline src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: Is this the correct indentation?

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx [hunk 3]
```diff
@@ -330,12 +331,12 @@ class ComparisonFilterUi extends React.Component<ComparisonFilterProps, Comparis
     this.setState({
       validateStatus: validationStateNew,
       filter
+    },            () => {
```
Comment (chrismayer) [inline src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: Is this the correct indentation?

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx (hunk 1 above)
Comment (ahennr) [reply src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: According to the current `tslint.config`, yes

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx (hunk 2 above)
Comment (ahennr) [reply src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: See above reply

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx (hunk 3 above)
Comment (ahennr) [reply src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: See above reply

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx (hunk 1 above)
Comment (chrismayer) [reply src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: OK, nice :wink:

File: src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx (hunk 3 above)
Comment (marcjansen) [reply src/Component/Filter/ComparisonFilter/ComparisonFilter.tsx]: it looks odd, tbh.

**Comment by ahennr** [comment]
Alright, I've fixed the indentation as proposed [here](https://stackoverflow.com/questions/50522824/react-callback-arguments-are-not-aligned)

**Pull request merged**